models/image_classification_ppresnet/demo.cpp

```diff
@@ -0,0 +1,1132 @@
+        class_id_list.resize(min(_topK, static_cast<int>(outputBlob.cols)));
+        vector<string> predicted_labels;
+        for (int class_id : class_id_list) 
+        {
+            predicted_labels.push_back(_labels[class_id]);
+        }
+        return predicted_labels;
+    }
+
+    void loadLabels() 
+    {
+        istringstream labelsStream(LABELS_IMAGENET_1K);
+        string line;
+        while (getline(labelsStream, line)) 
+        {
+            _labels.push_back(line);
+        }
+    }
+
+private:
+    Net _model;
+    int _topK;
+    vector<string> _labels;
+    const Size _inputSize = Size(224, 224);
+    const Scalar _mean = Scalar(0.485, 0.456, 0.406);
+    const Scalar _std = Scalar(0.229, 0.224, 0.225);
+    string _inputName = "";
+    string _outputName = "save_infer_model/scale_0.tmp_0";
+};
+
+const vector<vector<int>> backend_target_pairs = 
+{
+    {DNN_BACKEND_OPENCV, DNN_TARGET_CPU},
+    {DNN_BACKEND_CUDA, DNN_TARGET_CUDA},
+    {DNN_BACKEND_CUDA, DNN_TARGET_CUDA_FP16},
+    {DNN_BACKEND_TIMVX, DNN_TARGET_NPU},
+    {DNN_BACKEND_CANN, DNN_TARGET_NPU}
+};
+
+int main(int argc, char** argv) 
+{
+    CommandLineParser parser(argc, argv,
+        "{ input i               |                                               | Set input path to a certain image, omit if using camera.}"
+        "{ model m               | image_classification_ppresnet50_2022jan.onnx  | Set model path.}"
+        "{ top_k k               | 1                                             | Get top k predictions.}"
+        "{ backend_target bt     | 0                                             | Choose one of computation backends: "
+        "0: (default) OpenCV implementation + CPU, "
+        "1: CUDA + GPU (CUDA), "
+        "2: CUDA + GPU (CUDA FP16), "
+        "3: TIM-VX + NPU, "
+        "4: CANN + NPU}");
+
+    string inputPath = parser.get<string>("input");
+    string modelPath = parser.get<string>("model");
+    int backendTarget = parser.get<int>("backend_target");
+    int topK = parser.get<int>("top_k");
+
+    int backendId = backend_target_pairs[backendTarget][0];
+    int targetId = backend_target_pairs[backendTarget][1];
+
+    PPResNet model(modelPath, topK, backendId, targetId);
+
+    // Read image and get a 224x224 crop from a 256x256 resized
+    Mat image = imread(inputPath);
+    cvtColor(image, image, COLOR_BGR2RGB);
+    resize(image, image, Size(256, 256));
+    image = image(Rect(16, 16, 224, 224));
+
+    // Inference
+    auto predictions = model.infer(image);
+
+    // Print result
+    if (topK == 1)
+    {
+        cout << "Predicted Label: " << predictions[0] << endl;
+    }
+    else 
+    {
+        cout << "Predicted Top-K Labels (in decreasing confidence): " << endl;
+        for (size_t i = 0; i < predictions.size(); ++i) 
+        {
+            cout << "(" << i+1 << ") " << predictions[i] << endl;
+        }
+    }
+
+    return 0;
+}
+
+string LABELS_IMAGENET_1K = "tench\n"
```

Comment:
How about using `std::vector<std::string>`? In this case we do not need to duplicate `\n` for each class and we do not need to re-load this in `loadLabels`.

Reply:
@fengyuentau  That's a great idea. Not sure why I didn't think of that :sweat_smile: I've updated it.